packages/vats/test/upgrading/test-upgrade-vats.js

```diff
@@ -0,0 +1,56 @@
+import { test as anyTest } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { buildVatController } from '@agoric/swingset-vat';
+
+/** @type {import('ava').TestFn<Awaited<ReturnType<typeof makeTestContext>>>} */
+const test = anyTest;
+
+/**
+ * WARNING: uses ambient authority
+ * of import.meta.url
+ */
+const makeTestContext = async () => {
+  const bfile = name => new URL(name, import.meta.url).pathname;
+  return { bfile };
+};
+
+test.before(async t => {
+  t.context = await makeTestContext();
+});
+
+test('upgrade vat-board', async t => {
+  const { bfile } = t.context;
+
+  /** @type {SwingSetConfig} */
+  const config = harden({
+    defaultManagerType: 'xs-worker',
+    bootstrap: 'bootstrap',
+    vats: {
+      bootstrap: { sourceSpec: bfile('./bootstrapVatUpgrade.js') },
+    },
+    bundles: {
+      board: { sourceSpec: bfile('../../src/vat-board.js') },
+    },
+  });
+  // console.debug('config', JSON.stringify(config, null, 2));
+
+  const c = await buildVatController(config);
+  c.pinVatRoot('bootstrap');
+  await c.run();
+
+  const run = async (name, args = []) => {
+    assert(Array.isArray(args));
+    const kpid = c.queueToVatRoot('bootstrap', name, args);
+    await c.run();
+    const status = c.kpStatus(kpid);
+    const capdata = c.kpResolution(kpid);
+    return [status, capdata];
+  };
```

Comment:
out of scope but I imagine this snippet will be copied a lot. we might soon want more upgrade test supports from swingset-vat.